src/coreclr/tools/Common/InstructionSetHelpers.cs

```diff
@@ -93,16 +93,33 @@ internal static partial class Helpers
                 optimisticInstructionSetSupportBuilder.AddSupportedInstructionSet("movbe");
                 optimisticInstructionSetSupportBuilder.AddSupportedInstructionSet("popcnt");
                 optimisticInstructionSetSupportBuilder.AddSupportedInstructionSet("lzcnt");
+                optimisticInstructionSetSupportBuilder.AddSupportedInstructionSet("serialize");
 
                 // If AVX was enabled, we can opportunistically enable instruction sets which use the VEX encodings
                 Debug.Assert(InstructionSet.X64_AVX == InstructionSet.X86_AVX);
                 if (supportedInstructionSet.HasInstructionSet(InstructionSet.X64_AVX))
                 {
+                    optimisticInstructionSetSupportBuilder.AddSupportedInstructionSet("avx2");
```

Comment:
If we make this change we need to ensure that code generation when avx, but not avx2 is specified on the command line does not put us in a place where use of Vector<T> intrinsics doesn't works incorrectly. I'll need to examine an r2rdump of this case.

Reply:
Shouldn't that be enforced by `InstructionSet_VectorT256` only being set when `AVX2` is supported and the corresponding query for `Vector<T>` size being tied to `InstructionSet_VectorT256` (and other versions)?